### ISSUE TITLE
SUGGESTION: Implement a 'DOS_Sleep' function?

### ISSUE THREAD

**Issue by lighth7015** [teefoss/textmode#2]
Hi!

I recently stumbled across your project, and it works fantastically; I'm working on a project that replicates BIOS-like behavior, but I'm running into an issue where I can't cause my app to "sleep" for 2 or 3 seconds before refreshing the display.

Any ideas how I could go about doing that?

**Comment by teefoss** [comment]
I can add that at some point! In the meantime you should be able to use SDL_Delay, which takes a millisecond value.

**Comment by lighth7015** [comment]
If I do that, the application ends up freezing up, so what I ended up doing was something like this,
```c
int tick = 0;

int KdGetTickCount() {
    return tick;
}

int KdDispatch(int delay) {
     int cond = 0;

     if ((tick + 1) == delay) {
          tick = 0, cond = 1;
     }

     return cond;
}
```

this way, I can call the function on multiple passes and return either 0 or 1, depending on whether I want the function to signal to the rest of the application that I want it to continue.

My use-case is basically a full-screen BIOS-type UI, with a "copyright" splash screen that behaves as if it's a wizard.

**Comment by teefoss** [comment]
Glad you found a solution! Both SDL_Delay and the standard library function 'sleep' do suspend the program. If I add something, it'll be along those lines.

I reread your first message, you're looking to have the program run uninterrupted, but only refresh periodically? Ideally, things would work by not refreshing at all until something changes, like the cursor blinks, or there's user input (causing a selection highlight to change, for example). At some point, I'll be revamping things to get it to work like that.

A couple thoughts for now: If your question is to do with slow performance related to continuous refreshing, definitely avoid using DOS_RenderChar and DOS_RenderString, especially for a full screen's worth of text - they're pretty slow. Use a DOS_Console or 'screen' via DOS_InitScreen. Also, DOS_LimitFrameRate will slow things down a little.

**Comment by lighth7015** [comment]
Yes, I'm looking to "delay" the app without calling `sleep` under the hood.
